smartsim/experiment.py

```diff
@@ -184,6 +188,15 @@ def start(
 
         :type kill_on_interrupt: bool, optional
         """
+
```

Comment:
Are we sure that only a colocated db should get an identifier? Can we give them all one so it's more reliable?

Is the ID really just an index? How is the ID tied to a specific instance? Doesn't this require that all subsequent actions must operate on a list in the same sort order as the args to this particular method? Is that reliable?

I think there is probably a more appropriate place/way to do this. It feels odd that we're modifying run_settings in the start of launch, for only a subset of kwargs. Can't we avoid this check and let the manifest get built but modify that to correlate an ID with a specific instance?

Reply:
regular databases are also given db-ids. 
I think i was confused with how colocated models are started, you are right, this should be done differently/elsewhere

Reply:
Marking as resolved as it looks like it does not apply to current status of code

Reply:
Agreed, I think this comment no longer valid in current implementation